### PR TITLE
Refactoring the Transaction class.

### PR DESCRIPTION
**Pull request by mj4444ru**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | 

Related PRs:

- https://github.com/yiisoft/db-sqlite/pull/44
- https://github.com/yiisoft/db-pgsql/pull/51

### CODE REVIEW

File: src/Transaction/Transaction.php [hunk 1]
```diff
@@ -46,33 +45,37 @@ class Transaction
      * {@see http://en.wikipedia.org/wiki/Isolation_%28database_systems%29#Isolation_levels}
      */
     public const READ_UNCOMMITTED = 'READ UNCOMMITTED';
+    public const TIL_READ_UNCOMMITTED = 'READ UNCOMMITTED';
```
[the remark this answers is not in the document]
Comment (mj4444ru) [reply src/Transaction/Transaction.php:48]: When we create a constant for the `status` field, we call it `STATUS_ACTIVE`, but not `ACTIVE`.
The `SERIALIZABLE` style constants imported into the local scope can be confusing to me.
**TIL** - stands for **T**ransaction **I**solation **L**evel.

This is just a suggestion. It is possible to return constants without a prefix.

[the remark this answers is not in the document]
Comment (mj4444ru) [reply src/Transaction/Transaction.php:48]: When we create a constant for the `status` field, we call it` STATUS_ACTIVE`, but not ʻACTIVE`.
**TIL** stands for **T**ransaction **I**solation **L**evel.

**TIL_** is probably a bad option for a prefix. Can be changed to **IL_** (**I**solation **L**evel), or **LEVEL_**.

This is just a suggestion. It is possible to return constants without a prefix.

[the remark this answers is not in the document]
Comment (samdark) [reply src/Transaction/Transaction.php:48]: OK but there are already constants for the purpose, aren't there?

[the remark this answers is not in the document]
Comment (mj4444ru) [reply src/Transaction/Transaction.php:48]: I thought that they can be left for compatibility with yii2 and marked as deprecated), while in yii3 to encourage the use of new constants.

Your opinion?

[the remark this answers is not in the document]
Comment (samdark) [reply src/Transaction/Transaction.php:48]: We don't need to take care compatibiltiy too much and definitely don't want deprecated stuff in Yii 3 v1.0.

[the remark this answers is not in the document]
Comment (mj4444ru) [reply src/Transaction/Transaction.php:48]: In this case, I suggest choosing a prefix between **IL_** and **LEVEL**. And remove constants without a prefix.

[the remark this answers is not in the document]
Comment (samdark) [reply src/Transaction/Transaction.php:48]: LEVEL is better.

File: src/Transaction/Transaction.php [hunk 2]
```diff
@@ -45,34 +44,34 @@ class Transaction
      *
      * {@see http://en.wikipedia.org/wiki/Isolation_%28database_systems%29#Isolation_levels}
      */
-    public const READ_UNCOMMITTED = 'READ UNCOMMITTED';
+    public const LEVEL_READ_UNCOMMITTED = 'READ UNCOMMITTED';
```
Comment (samdark) [inline src/Transaction/Transaction.php:47]: It's used in ConnectionTest. Should be renamed there as well.

Comment (samdark) [reply src/Transaction/Transaction.php:47]: It's used in sqlite and pgsql driver packages.

**Review by samdark — CHANGES_REQUESTED** [review]

**Comment by terabytesoftw** [comment]
closed due to inactivity

**Pull request closed without merging**